src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -772,6 +821,16 @@ public async Task<HttpResponseMessage> SendWithRetryAsync(HttpRequestMessage req
                     // Eat exception and try again.
                     continue;
                 }
+                catch (HttpRequestException e) when (!isNewConnection && e.AllowRetry == RequestRetryType.RetryOnNewConnection)
```

Comment:
Instead of introducing a new enum value, can you use `RetryOnSameOrNextProxy`? It appears to have the desired effect, and will be more efficient to check fewer exception filters.